src/main/java/practice/StreamPractice.java

```diff
@@ -1,80 +1,73 @@
 package practice;
 
-import java.util.Collections;
+import java.util.Arrays;
+import java.util.Comparator;
 import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
 import model.Candidate;
+import model.Cat;
 import model.Person;
 
 public class StreamPractice {
-    /**
-     * Given list of strings where each element contains 1+ numbers:
-     * input = {"5,30,100", "0,22,7", ...}
-     * return min integer value. One more thing - we're interested in even numbers.
-     * If there is no needed data throw RuntimeException with message
-     * "Can't get min value from list: < Here is our input 'numbers' >"
-     */
+    private static final int ZERO_NUMBER = 0;
+    private static final int SUBTRACT_NUMBER = 1;
+    private static final int CHECKING_NUMBER = 2;
+
     public int findMinEvenNumber(List<String> numbers) {
-        return 0;
+        return Integer.parseInt(numbers.stream()
+                .map(s -> s.split(","))
+                .flatMap(Arrays::stream)
+                .filter(n -> Integer.parseInt(n) % CHECKING_NUMBER == 0)
+                .min(Comparator.comparingInt(Integer::parseInt))
+                .orElseThrow(() ->
+                        new RuntimeException("Can't get min value from list:" + numbers)));
     }
 
-    /**
-     * Given a List of Integer numbers,
-     * return the average of all odd numbers from the list or throw NoSuchElementException.
-     * But before that subtract 1 from each element on an odd position (having the odd index).
-     */
     public Double getOddNumsAverage(List<Integer> numbers) {
-        return 0D;
+        return IntStream.range(ZERO_NUMBER, numbers.size())
+                .map(i -> i % CHECKING_NUMBER != ZERO_NUMBER
+                        ? numbers.get(i) - SUBTRACT_NUMBER : numbers.get(i))
+                .filter(n -> n % CHECKING_NUMBER != ZERO_NUMBER)
```

Comment:
```suggestion
                .map(i -> i % 2 != 0 ? numbers.get(i) - SUBTRACT_NUMBER : numbers.get(i))
                .filter(n -> n % 2 != 0)
```
same thing with constants